.pre-commit-config.yaml

```diff
@@ -0,0 +1,50 @@
+repos:
+  - repo: local
+    hooks:
+      #Formatting
+      - id: prettier
+        name: Prettier (Code formatting)
+        entry: npm run format
+        language: node
+        files: \.(js|jsx|ts|tsx)$
+
+      - id: black
+        name: Black (Python code formatter)
+        entry: black --line-length=88 .
+        language: system
+        files: data/
+        types: [python]
+
+      # Linting
+      - id: eslint
+        name: ESLint (JavaScript linting)
+        entry: npm run lint
+        language: node
+        files: \.(js|jsx|ts|tsx)$
+
+      - id: flake8
+        name: Flake8 (Python linting)
+        entry: flake8
+        language: system
+        files: data/
+        types: [python]
+
+      # Dependency checks
+      - id: npm-audit
+        name: NPM Audit (Check Node.js dependencies)
+        entry: npm audit fix
+        language: node
+
+      - id: check
+        name: Check (Check Python dependencies)
+        entry: pipenv check
+        language: system
+        files: data/
+
+      # Commit message linting
+      - id: commitlint
```

Comment:
What's the team buy-in for adopting conventional commits? I've had trouble in the past introducing pre-commit conventional commit checks. The trouble was partially due to the checks interfering with the workflow of committing early and often as part of their workflow, then rebasing with a nicer message. The fix for that was checking commits pre-push rather than pre-commit.  The other problem was figuring out what's allowed in the (scope) part of a conventional commit. 

<details><summary>(Semi-related shameless plug: I've written several tools to help write conventional commits).</summary>

- [`git-cc`](https://github.com/SKalt/git-cc), a terminal UI that fits into a command-line git workflow
- [`cconvention`](https://github.com/skalt/cconvention), a language server and [vscode extension](https://marketplace.visualstudio.com/items?itemName=kalt-dot-cloud.git-conventional-commit-ls-basic) for conventional commits in your `$EDITOR`. 

</details>